### DISCUSSION TITLE
[Integration] Single-Source Hardware-Software Codesign

### DISCUSSION THREAD

**Discussion by tinebp** [cucapra/latte21#44] [Papers]
Authors: Blaise Tine, Sudhakar Yalamanchili, Hyesoon Kim
Paper PDF: https://capra.cs.cornell.edu/latte21/paper/24.pdf
Talk Recording: https://www.youtube.com/watch?v=WxpZwn6PafY

**Comment by gussmith23** [comment]
I love the idea behind this! I would love to have a discussion on _which language_ should be the single-source language. C++ makes sense from a pragmatic angle, as it's absolutely everywhere. Yet its generality also may be a pain point. I imagine more specialized languages -- which could be embedded into C++! -- might make this single-source vision more feasible. Thoughts?

**Reply by tinebp** [reply]
The long-term goal behind the Cash project was to explore an actual language extension to C++, similar to OpenCL or CUDA that remains compatible with C++ headers and syntax.
Another option is to follow the OpenMP or OpenACC solution with directives where hardware blocks are simply embedded inside C++.
